jdaviz/core/helpers.py

```diff
@@ -428,7 +428,23 @@ def _handle_display_units(data, use_display_units):
                     flux_unit = self.app._get_display_unit('flux')
                     # TODO: any other attributes (meta, wcs, etc)?
                     # TODO: implement uncertainty.to upstream
-                    new_uncert = data.uncertainty.__class__(data.uncertainty.quantity.to(flux_unit)) if data.uncertainty is not None else None  # noqa
+                    uncertainty = data.uncertainty
+                    if uncertainty is not None:
+                        # convert the uncertainties to StdDevUncertainties, since
+                        # that is assumed in a few places in jdaviz:
+                        if uncertainty.unit is None:
+                            uncertainty = uncertainty.unit = data.flux.unit
```

Comment:
```suggestion
                            uncertainty.unit = data.flux.unit
```
We don't want to set the whole uncertainty to the unit, looks like a typo.

Reply:
Thank you!